Fail to launch jobs on Amazon EC2 with master-IP

I try to run several [spark-bench](https://github.com/SparkTC/spark-bench) benchmarks on EC2 launched by flintrock.

In the `standalone` mode of spark, I have to configure the IP address of EC2 machine as the "master" of spark.

The console and .conf is below:

```
[ec2-user@ip-172-31-15-16 new-spark-bench_2.1.1]$ ./bin/spark-bench.sh examples/kmeans.conf
 *** SPARK-SUBMIT: [/home/ec2-user/spark/bin/spark-submit, --class, com.ibm.sparktc.sparkbench.cli.CLIKickoff, --master, spark://34.215.233.221:7077, /home/ec2-user/new-spark-bench_2.1.1/lib/spark-bench-2.1.1_0.2.2-RELEASE.jar, {"spark-bench":{"spark-submit-config":[{"spark-args":{"master":"spark://34.215.233.221:7077"},"workload-suites":[{"benchmark-output":"console","descr":"datagen kmeans","workloads":[{"cols":4,"name":"data-generation-kmeans","output":"file:///tmp/kmeans-data.csv","parititions":32,"rows":10}]},{"benchmark-output":"console","descr":"run kmeans","workloads":[{"input":"file:///tmp/KMeen.csv","name":"kmeans"}]}]}]}}]
Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
17/11/28 09:04:11 INFO SparkContext: Running Spark version 2.2.0
17/11/28 09:04:11 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
17/11/28 09:04:12 INFO SparkContext: Submitted application: com.ibm.sparktc.sparkbench.cli.CLIKickoff
17/11/28 09:04:12 INFO SecurityManager: Changing view acls to: ec2-user
17/11/28 09:04:12 INFO SecurityManager: Changing modify acls to: ec2-user
17/11/28 09:04:12 INFO SecurityManager: Changing view acls groups to:
17/11/28 09:04:12 INFO SecurityManager: Changing modify acls groups to:
17/11/28 09:04:12 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(ec2-user); groups with view permissions: Set(); users  with modify permissions: Set(ec2-user); groups with modify permissions: Set()
17/11/28 09:04:12 INFO Utils: Successfully started service 'sparkDriver' on port 37999.
17/11/28 09:04:12 INFO SparkEnv: Registering MapOutputTracker
17/11/28 09:04:12 INFO SparkEnv: Registering BlockManagerMaster
17/11/28 09:04:12 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
17/11/28 09:04:12 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
17/11/28 09:04:12 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-f402d6cf-fed7-442c-abe5-a9ae2167051d
17/11/28 09:04:12 INFO MemoryStore: MemoryStore started with capacity 366.3 MB
17/11/28 09:04:12 INFO SparkEnv: Registering OutputCommitCoordinator
17/11/28 09:04:13 INFO Utils: Successfully started service 'SparkUI' on port 4040.
17/11/28 09:04:13 INFO SparkUI: Bound SparkUI to 0.0.0.0, and started at http://172.31.15.16:4040
17/11/28 09:04:13 INFO SparkContext: Added JAR file:/home/ec2-user/new-spark-bench_2.1.1/lib/spark-bench-2.1.1_0.2.2-RELEASE.jar at spark://172.31.15.16:37999/jars/spark-bench-2.1.1_0.2.2-RELEASE.jar with timestamp 1511859853321
17/11/28 09:04:13 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://34.215.233.221:7077...
17/11/28 09:04:33 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://34.215.233.221:7077...
17/11/28 09:04:53 INFO StandaloneAppClient$ClientEndpoint: Connecting to master 
```

```
17/11/28 15:45:36 INFO SecurityManager: Changing view acls to: ec2-user
17/11/28 15:45:36 INFO SecurityManager: Changing modify acls to: ec2-user
17/11/28 15:45:36 INFO SecurityManager: Changing view acls groups to:
17/11/28 15:45:36 INFO SecurityManager: Changing modify acls groups to:
17/11/28 15:45:36 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(ec2-user); groups with view permissions: Set(); users  with modify permissions: Set(ec2-user); groups with modify permissions: Set()
17/11/28 15:45:36 INFO Utils: Successfully started service 'sparkDriver' on port 32937.
17/11/28 15:45:36 INFO SparkEnv: Registering MapOutputTracker
17/11/28 15:45:36 INFO SparkEnv: Registering BlockManagerMaster
17/11/28 15:45:36 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
17/11/28 15:45:36 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
17/11/28 15:45:36 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-b05bab88-e040-4614-9665-bbeeea5f5c94
17/11/28 15:45:36 INFO MemoryStore: MemoryStore started with capacity 366.3 MB
17/11/28 15:45:37 INFO SparkEnv: Registering OutputCommitCoordinator
17/11/28 15:45:37 INFO Utils: Successfully started service 'SparkUI' on port 4040.
17/11/28 15:45:37 INFO SparkUI: Bound SparkUI to 0.0.0.0, and started at http://172.31.6.28:4040
17/11/28 15:45:37 INFO SparkContext: Added file file:/home/ec2-user/spark/examples/src/main/python/pi.py at spark://172.31.6.28:32937/files/pi.py with timestamp 1511883937418
17/11/28 15:45:37 INFO Utils: Copying /home/ec2-user/spark/examples/src/main/python/pi.py to /tmp/spark-eb3b751f-7e90-49d2-b1be-ab5fa1fd4eb1/userFiles-e6575cb1-3e15-4f5b-b3e0-3d707eea6e9a/pi.py
17/11/28 15:45:37 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://35.162.130.151:7077...
17/11/28 15:45:37 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master 35.162.130.151:7077
org.apache.spark.SparkException: Exception thrown in awaitResult
	at org.apache.spark.rpc.RpcTimeout$$anonfun$1.applyOrElse(RpcTimeout.scala:77)
	at org.apache.spark.rpc.RpcTimeout$$anonfun$1.applyOrElse(RpcTimeout.scala:75)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
	at org.apache.spark.rpc.RpcTimeout$$anonfun$addMessageIfTimeout$1.applyOrElse(RpcTimeout.scala:59)
	at org.apache.spark.rpc.RpcTimeout$$anonfun$addMessageIfTimeout$1.applyOrElse(RpcTimeout.scala:59)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:83)
	at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:100)
	at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:108)
	at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anonfun$tryRegisterAllMasters$1$$anon$1.run(StandaloneAppClient.scala:106)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Failed to connect to /35.162.130.151:7077
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:232)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:182)
	at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:197)
	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:194)
	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:190)
	... 4 more
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /35.162.130.151:7077
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	... 1 more
17/11/28 15:45:57 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://35.162.130.151:7077...
17/11/28 15:45:57 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master 35.162.130.151:7077
org.apache.spark.SparkException: Exception thrown in awaitResult
	at org.apache.spark.rpc.RpcTimeout$$anonfun$1.applyOrElse(RpcTimeout.scala:77)
	at org.apache.spark.rpc.RpcTimeout$$anonfun$1.applyOrElse(RpcTimeout.scala:75)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
	at org.apache.spark.rpc.RpcTimeout$$anonfun$addMessageIfTimeout$1.applyOrElse(RpcTimeout.scala:59)
	at org.apache.spark.rpc.RpcTimeout$$anonfun$addMessageIfTimeout$1.applyOrElse(RpcTimeout.scala:59)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:83)
	at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:100)
	at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:108)
	at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anonfun$tryRegisterAllMasters$1$$anon$1.run(StandaloneAppClient.scala:106)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Failed to connect to /35.162.130.151:7077
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:232)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:182)
	at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:197)
	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:194)
	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:190)
	... 4 more
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /35.162.130.151:7077
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	... 1 more
17/11/28 15:46:17 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://35.162.130.151:7077...
17/11/28 15:46:17 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master 35.162.130.151:7077
org.apache.spark.SparkException: Exception thrown in awaitResult
	at org.apache.spark.rpc.RpcTimeout$$anonfun$1.applyOrElse(RpcTimeout.scala:77)
	at org.apache.spark.rpc.RpcTimeout$$anonfun$1.applyOrElse(RpcTimeout.scala:75)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
	at org.apache.spark.rpc.RpcTimeout$$anonfun$addMessageIfTimeout$1.applyOrElse(RpcTimeout.scala:59)
	at org.apache.spark.rpc.RpcTimeout$$anonfun$addMessageIfTimeout$1.applyOrElse(RpcTimeout.scala:59)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:83)
	at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:100)
	at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:108)
	at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anonfun$tryRegisterAllMasters$1$$anon$1.run(StandaloneAppClient.scala:106)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Failed to connect to /35.162.130.151:7077
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:232)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:182)
	at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:197)
	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:194)
	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:190)
	... 4 more
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /35.162.130.151:7077
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	... 1 more
17/11/28 15:46:37 ERROR StandaloneSchedulerBackend: Application has been killed. Reason: All masters are unresponsive! Giving up.
17/11/28 15:46:37 WARN StandaloneSchedulerBackend: Application ID is not initialized yet.
17/11/28 15:46:37 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 35355.
17/11/28 15:46:37 INFO NettyBlockTransferService: Server created on 172.31.6.28:35355
17/11/28 15:46:37 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
17/11/28 15:46:37 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 172.31.6.28, 35355, None)
17/11/28 15:46:37 INFO BlockManagerMasterEndpoint: Registering block manager 172.31.6.28:35355 with 366.3 MB RAM, BlockManagerId(driver, 172.31.6.28, 35355, None)
17/11/28 15:46:37 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 172.31.6.28, 35355, None)
17/11/28 15:46:37 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 172.31.6.28, 35355, None)
17/11/28 15:46:37 INFO SparkUI: Stopped Spark web UI at http://172.31.6.28:4040
17/11/28 15:46:37 INFO StandaloneSchedulerBackend: Shutting down all executors
17/11/28 15:46:37 INFO CoarseGrainedSchedulerBackend$DriverEndpoint: Asking each executor to shut down
17/11/28 15:46:37 WARN StandaloneAppClient$ClientEndpoint: Drop UnregisterApplication(null) because has not yet connected to master
17/11/28 15:46:37 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
17/11/28 15:46:37 INFO MemoryStore: MemoryStore cleared
17/11/28 15:46:37 INFO BlockManager: BlockManager stopped
17/11/28 15:46:37 INFO BlockManagerMaster: BlockManagerMaster stopped
17/11/28 15:46:37 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
17/11/28 15:46:37 INFO SparkContext: Successfully stopped SparkContext
17/11/28 15:46:37 ERROR SparkContext: Error initializing SparkContext.
java.lang.IllegalArgumentException: requirement failed: Can only call getServletHandlers on a running MetricsSystem
	at scala.Predef$.require(Predef.scala:224)
	at org.apache.spark.metrics.MetricsSystem.getServletHandlers(MetricsSystem.scala:91)
	at org.apache.spark.SparkContext.<init>(SparkContext.scala:524)
	at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:236)
	at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
	at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
	at py4j.GatewayConnection.run(GatewayConnection.java:214)
	at java.lang.Thread.run(Thread.java:748)
17/11/28 15:46:37 INFO SparkContext: SparkContext already stopped.
Traceback (most recent call last):
  File "/home/ec2-user/spark/examples/src/main/python/pi.py", line 32, in <module>
    .appName("PythonPi")\
  File "/home/ec2-user/spark/python/lib/pyspark.zip/pyspark/sql/session.py", line 169, in getOrCreate
  File "/home/ec2-user/spark/python/lib/pyspark.zip/pyspark/context.py", line 310, in getOrCreate
  File "/home/ec2-user/spark/python/lib/pyspark.zip/pyspark/context.py", line 118, in __init__
  File "/home/ec2-user/spark/python/lib/pyspark.zip/pyspark/context.py", line 182, in _do_init
  File "/home/ec2-user/spark/python/lib/pyspark.zip/pyspark/context.py", line 249, in _initialize_context
  File "/home/ec2-user/spark/python/lib/py4j-0.10.4-src.zip/py4j/java_gateway.py", line 1401, in __call__
  File "/home/ec2-user/spark/python/lib/py4j-0.10.4-src.zip/py4j/protocol.py", line 319, in get_return_value
py4j.protocol.Py4JJavaError: An error occurred while calling None.org.apache.spark.api.java.JavaSparkContext.
: java.lang.IllegalArgumentException: requirement failed: Can only call getServletHandlers on a running MetricsSystem
	at scala.Predef$.require(Predef.scala:224)
	at org.apache.spark.metrics.MetricsSystem.getServletHandlers(MetricsSystem.scala:91)
	at org.apache.spark.SparkContext.<init>(SparkContext.scala:524)
	at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:236)
	at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
	at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
	at py4j.GatewayConnection.run(GatewayConnection.java:214)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/flintrock/flintrock.py
+++ b/flintrock/flintrock.py
@@ -329,6 +329,11 @@ def launch(
             '--spark-git-commit'],
         scope=locals())
     option_requires(
+        option='--install-spark',
+        requires_all=[
+            '--hdfs-version'],
+        scope=locals())
+    option_requires(
         option='--provider',
         conditional_value='ec2',
         requires_all=[

--- a/flintrock/flintrock.py
+++ b/flintrock/flintrock.py
@@ -208,12 +208,7 @@ def cli(cli_context, config, provider, debug):
 @click.argument('cluster-name')
 @click.option('--num-slaves', type=click.IntRange(min=1), required=True)
 @click.option('--install-hdfs/--no-install-hdfs', default=False)
-@click.option('--hdfs-version',
-              # Don't set a default here because it may conflict with
-              # the config file.
-              # See: https://github.com/nchammas/flintrock/issues/190
-              # default=
-              )
+@click.option('--hdfs-version', default='2.7.4')
 @click.option('--hdfs-download-source',
               help="URL to download Hadoop from.",
               default='http://www.apache.org/dyn/closer.lua/hadoop/common/hadoop-{v}/hadoop-{v}.tar.gz?as_json',
@@ -222,8 +217,8 @@ def cli(cli_context, config, provider, debug):
 @click.option('--spark-executor-instances', default=1,
               help="How many executor instances per worker.")
 @click.option('--spark-version',
-              # Don't set a default here because it may conflict with
-              # the config file.
+              # Don't set a default here because it will conflict with
+              # the config file if the git commit is set.
               # See: https://github.com/nchammas/flintrock/issues/190
               # default=,
               help="Spark release version to install.")

